mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/JobClustersManagerActor.java

```diff
@@ -537,7 +538,15 @@ public void onWorkerEvent(WorkerEvent workerEvent) {
             if(!JobHelper.isTerminalWorkerEvent(workerEvent)) {
                 logger.warn("Event from Worker {} for a cluster {} that no longer exists. Terminate worker", workerEvent, workerEvent.getWorkerId().getJobCluster());
                 Optional<String> host = JobHelper.getWorkerHostFromWorkerEvent(workerEvent);
-                mantisScheduler.unscheduleAndTerminateWorker(workerEvent.getWorkerId(), host);
+                Optional<JobDefinition> archivedJobDefinition =
+                    jobClusterInfoManager.getArchivedJobDefinition(workerEvent.getWorkerId().getJobId());
```

Comment:
@Andyz26 can you check if this is valid?

Reply:
lgtm